dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java

```diff
@@ -109,24 +114,52 @@ private PredicateBuilder initiatorPredicateBuilder(long initiatorUserId) {
 
     private PredicateBuilder accessPredicateBuilder(User loggedInUser) {
         return (cb, event) -> {
-            // Calculate a predicate representing public access.
-            // Currently, this is all events that don't refer to a "categorizer" organization.
-            Join<Event, Organization> organization = event.join("organization", JoinType.LEFT);
-            Predicate publicPredicate = cb.or(cb.isFalse(organization.get("categorizer")), cb.isNull(organization));
-            // Calculate a predicate representing extra access due to the user's identity.
-            Predicate extraPredicate;
             if (loggedInUser != null && (loggedInUser.getIsAdmin() || loggedInUser.isCurator())) {
-                // Admin or curator, able to access everything.
-                extraPredicate = cb.conjunction();
-            } else if (loggedInUser != null) {
-                // Logged in, able to access events that the user initiated.
-                extraPredicate = event.get("initiatorUser").in(loggedInUser.getId());
+                // An admin or curator is able to access everything.
+                return cb.conjunction();
+            }
+            // Calculate a predicate representing public access, which, currently, is all events that:
+            // 1) Either:
+            // a) Don't refer to an entry
+            // b) Refer to a published entry, or
+            // c) Are of type PUBLISH_ENTRY or UNPUBLISH_ENTRY or COLLECTION-related events
+            // and
+            // 2) Don't refer to a categorizer organization.
+            Join<Event, Tool> tool = event.join("tool", JoinType.LEFT);
+            Join<Event, BioWorkflow> workflow = event.join("workflow", JoinType.LEFT);
+            Join<Event, AppTool> apptool = event.join("apptool", JoinType.LEFT);
+            Join<Event, Service> service = event.join("service", JoinType.LEFT);
+            Join<Event, Notebook> notebook = event.join("notebook", JoinType.LEFT);
+            Predicate noEntryPredicate = cb.and(
+                cb.isNull(tool),
+                cb.isNull(workflow),
+                cb.isNull(apptool),
+                cb.isNull(service),
+                cb.isNull(notebook)
+            );
+            Predicate publishedEntryPredicate = cb.or(
+                cb.isTrue(tool.get("isPublished")),
+                cb.isTrue(workflow.get("isPublished")),
```

Comment:
Nit: this 5x repetition of `isPublished`; you could have a typo in one of them :) (You don't but you get the idea).

Reply:
This is the reported code smell btw, same reasoning